### PR TITLE
issue/3163 Added _hasItemScoring

### PR DESCRIPTION
**Pull request by oliverfoster**
Part of https://github.com/adaptlearning/adapt_framework/issues/3163

### Added
* `_hasItemScoring` support

Required https://github.com/adaptlearning/adapt_framework/pull/3172

### CODE REVIEW

File: properties.schema [hunk 1]
```diff
@@ -102,6 +102,15 @@
             "validators": [],
             "help": "When 'Selectable Items' is set to 1, this can be used to give the user feedback specific to the answer they selected - if incorrect.",
             "translatable": true
+          },
+          "_score": {
+            "type": "number",
+            "required": false,
+            "default": 0,
+            "title": "Attempts",
```
Comment (tomgreenfield) [inline properties.schema:110]:
```suggestion
            "title": "Score",
```

File: README.md [hunk 2]
```diff
@@ -49,7 +49,9 @@ guide the learner’s interaction with the component.
 
 **\_isRandom** (boolean): Setting this value to `true` will cause the `_items` to appear in a random order each time the component is loaded. The default is `false`.   
 
-**\_questionWeight** (number): A number which reflects the significance of the question in relation to the other questions in the course. This number is used in calculations of the final score reported to the LMS.  
+**\_hasItemScoring** (boolean): When `false`, this question scores 0 for incorrect and 'Question Weight' for correct. When `true`, this question scores by summing the `_score` of the selected items.  
+
+**\_questionWeight** (number): When '_hasItemScoring' is `false`, this is the question score for a correct response. This number is used in calculations of the final score reported to the LMS.
```
Comment (tomgreenfield) [inline README.md:54]:
```suggestion
**\_questionWeight** (number): When `_hasItemScoring` is `false`, this is the question score for a correct response. This number is used in calculations of the final score reported to the LMS.  
```

File: README.md [hunk 3]
```diff
@@ -49,7 +49,9 @@ guide the learner’s interaction with the component.
 
 **\_isRandom** (boolean): Setting this value to `true` will cause the `_items` to appear in a random order each time the component is loaded. The default is `false`.   
 
-**\_questionWeight** (number): A number which reflects the significance of the question in relation to the other questions in the course. This number is used in calculations of the final score reported to the LMS.  
+**\_hasItemScoring** (boolean): When `false`, this question scores 0 for incorrect and 'Question Weight' for correct. When `true`, this question scores by summing the `_score` of the selected items.
```
Comment (tomgreenfield) [inline README.md:52]:
```suggestion
**\_hasItemScoring** (boolean): When `false`, this question scores 0 for incorrect and '`_questionWeight` for correct. When `true`, this question scores by summing the `_score` of the selected items.  
```

File: README.md [hunk 4]
```diff
@@ -69,6 +71,8 @@ guide the learner’s interaction with the component.
 
 >**feedback** (string): This attribute is used only when the value for **\_selectable** is set to `1` (i.e., radio button style questions). This text will be shown if the learner selects this item, and it is an incorrect answer. 
 
+>**\_score** (boolean): If 'Item Scoring' is `true`, when selected, item scores are summed to give the question score.
```
Comment (tomgreenfield) [inline README.md:74]:
```suggestion
>**\_score** (number): If `_hasItemScoring` is `true`, when selected, item scores are summed to give the question score.  
```

File: properties.schema [hunk 5]
```diff
@@ -165,14 +174,23 @@
       "validators": [],
       "help": "If disabled, recording the user's answer(s) to this question to cmi.interactions on the LMS will be disabled for this component only."
     },
+    "_hasItemScoring": {
+      "type": "boolean",
+      "required": false,
+      "default": false,
+      "title": "Item Scoring",
```
Comment (tomgreenfield) [inline properties.schema:181]:
```suggestion
      "title": "Enable item scoring",
```

File: properties.schema [hunk 6]
```diff
@@ -165,14 +174,23 @@
...
+    },
     "_questionWeight": {
       "type": "number",
       "required": false,
       "default": 1,
       "title": "Question Weight",
       "inputType": "Number",
       "validators": ["number"],
-      "help": "How much this question is worth"
+      "help": "When 'Item Scoring' is disabled, this is the question score for a correct response."
```
Comment (tomgreenfield) [inline properties.schema:193]:
```suggestion
      "help": "When item scoring is disabled, this is the question score for a correct response."
```

File: properties.schema [hunk 7]
```diff
@@ -102,6 +102,15 @@
...
             "translatable": true
+          },
+          "_score": {
+            "type": "number",
+            "required": false,
+            "default": 0,
+            "title": "Attempts",
+            "inputType": "Number",
+            "validators": ["required", "number"],
+            "help": "If 'Item Scoring' is enabled, when selected, item scores are summed to give the question score."
```
Comment (tomgreenfield) [inline properties.schema:113]:
```suggestion
            "help": "If item scoring is enabled, selected item scores are summed to give the question score."
```

**Review by tomgreenfield — CHANGES_REQUESTED** [review]

**Review by tomgreenfield — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by eleanor-heath — APPROVED** [review]

**Pull request merged**